### PR TITLE
Fix #118, will not return deleted info anymore

### LINKED ISSUE #118: Merak Network: The old network config info still in db after the network config got deleted
Reproduce Step:
1. Deploy a network config (1 vpc, 1 security group, 1 subnet) from scenario manager
2. Delete the network config from scenario manager 
3. Deploy a new network config again (1 vpc, 1 security group, 1 subnet) 

Result:
Merak network returns two vpcs, 2 security groups, 2 subnets

Merak network seems didn't clean up the db or it may use global variable to keep the network config in the memory.

```
{
    "data": {
        "returnMessage": "returnNetworkMessage Finished",
        "securityGroupIds": [
            "d0ff0ae2120f436ba4b6576a69ea916f",
            "7c2f29a850c24f50966ad391cc7dd391"
        ],
        "vpcs": [
            {
                "projectId": "123456789",
                "subnets": [
                    {
                        "numberVms": 100,
                        "subnetCidr": "10.0.1.0/20",
                        "subnetGw": "10.0.1.1",
                        "subnetId": "590d16bc-cfb6-4e55-9458-1086b54da56f"
                    }
                ],
                "tenantId": "123456789",
                "vpcId": "f15fa247-1c8c-4d22-bf74-232f1007c256"
            },
            {
                "projectId": "123456789",
                "subnets": [
                    {
                        "numberVms": 100,
                        "subnetCidr": "10.0.1.0/20",
                        "subnetGw": "10.0.1.1",
                        "subnetId": "dc5058cd-8bbb-45c2-b3e4-3164f00833ce"
                    }
                ],
                "tenantId": "123456789",
                "vpcId": "7886aa20-8974-445b-8d47-51b75a852c4d"
            }
        ]
    },
    "message": "Action successfully - DEPLOY on Network done",
    "status": "OK"
}
```

### PR DESCRIPTION
**Pull request by yanmo96**

### CODE REVIEW

**Comment by yanmo96** [comment]
Originally try to fix with channel, thinking maybe there leftover in there. But is good to have it to prevent later issue. https://github.com/futurewei-cloud/merak/blob/82e613d13f902b22bbf8fd064cd56db6c2ccc198/services/merak-network/grpc/service/service.go#L78 
Now the actual bug was in the network create, a global var was never cleared. That's fixed in this PR.
Have been tested locally, a new image was build yesterday for the .77 env. If that image perform as expected then this PR can be merged.

**Comment by cj-chung** [comment]
@yanmo96 Please verify it on .77

File: services/merak-network/activities/vnet_create_activity.go [hunk 1]
```diff
@@ -160,10 +160,12 @@ func doSg(sg *pb.InternalSecurityGroupInfo, sgID string, projectId string) (stri
...
 func VnetCreate(ctx context.Context, netConfigId string, network *pb.InternalNetworkInfo, wg *sync.WaitGroup, returnMessage chan *pb.ReturnNetworkMessage, projectId string) (*pb.ReturnNetworkMessage, error) {
 	log.Println("VnetCreate")
-	//defer wg.Done()
 	// TODO may want to separate bellow sections to different function, and use `go` and `wg` to improve overall speed
 	// TODO when do concurrent, need to keep in mind on how to control the number of concurrency
-	// Doing vpc and subnet
+
+	// clear the returnNetworkMessage
+	returnNetworkMessage.Vpcs = nil
+	returnNetworkMessage.SecurityGroupIds = nil
```
Comment (cj-chung) [inline services/merak-network/activities/vnet_create_activity.go:168]: @yanmo96 Do you think it's safe to have global variable for `reutrnNetworkMessage`, and return the same object from here?

Comment (yanmo96) [reply services/merak-network/activities/vnet_create_activity.go:168]: Good point, It was unnecessary to use a global var here. I will change it to local.

File: services/merak-network/activities/vnet_create_activity.go [hunk 2]
```diff
@@ -160,10 +151,15 @@ func doSg(sg *pb.InternalSecurityGroupInfo, sgID string, projectId string) (stri
 
 func VnetCreate(ctx context.Context, netConfigId string, network *pb.InternalNetworkInfo, wg *sync.WaitGroup, returnMessage chan *pb.ReturnNetworkMessage, projectId string) (*pb.ReturnNetworkMessage, error) {
 	log.Println("VnetCreate")
```
Comment (cj-chung) [inline services/merak-network/activities/vnet_create_activity.go:153]: @yanmo96 Did you use `ctx`, `network`, and `returnMessage` passing in parameters in this function?
If not, please remove them.

**Review by cj-chung — CHANGES_REQUESTED** [review]

**Comment by cj-chung** [comment]
@yanmo96 in line#203: `sgId := utils.GenUUID()`, why you need to generate uuid for sg? why not just let Alcor create one for you?

**Comment by cj-chung** [comment]
@yanmo96 in line#219: `err := doAttachRouter(routerId, subnetCiderIdMap[subnet], projectId)`, why you need to use `subnet_cider` to map `subnet_id`? I remember the input from scenario manager for router is `subnet_gw`.
Also, why the router didn't append to the `returnMessage`?

**Comment by yanmo96** [comment]
> 

I will double check if Alcor will give one or not

**Comment by yanmo96** [comment]
> doAttachRouter

It use the subnet cider to find the subnet id from the map. Then use the subnet id to attach it to router

**Comment by yanmo96** [comment]
Will have a new PR to add the router info and gateway info into it

**Comment by yanmo96** [comment]
> @yanmo96 in line#203: `sgId := utils.GenUUID()`, why you need to generate uuid for sg? why not just let Alcor create one for you?

I tried to create a security group without an ID, it returned to me without assign a UUID to it. Bellow is the return:
```
{
    "id": "",
    "project_id": "123456789",
    "tenant_id": "123456789",
    "name": "YM_sample_sg",
    "description": "sg Description",
    "security_group_rules": [
        {
            "id": "8f01055a-eea5-407f-85a1-e1fcee5680e4",
            "project_id": "",
            "tenant_id": "123456789",
            "name": "",
            "description": "",
            "security_group_id": "",
            "remote_group_id": null,
            "direction": "egress",
            "remote_ip_prefix": null,
            "protocol": null,
            "port_range_max": null,
            "port_range_min": null,
            "ethertype": "IPv4"
        },
        {
            "id": "cfbb758a-224d-42fc-8099-fa0e9df30969",
            "project_id": "",
            "tenant_id": "123456789",
            "name": "",
            "description": "",
            "security_group_id": "",
            "remote_group_id": null,
            "direction": "egress",
            "remote_ip_prefix": null,
            "protocol": null,
            "port_range_max": null,
            "port_range_min": null,
            "ethertype": "IPv6"
        }
    ],
    "create_at": "2022-09-19 23:13:58",
    "update_at": "2022-09-19 23:13:58"
}
```

**Comment by cj-chung** [comment]
> > @yanmo96 in line#203: `sgId := utils.GenUUID()`, why you need to generate uuid for sg? why not just let Alcor create one for you?
> 
> I tried to create a security group without an ID, it returned to me without assign a UUID to it. Bellow is the return:
> 
> ```
> {
>     "id": "",
>     "project_id": "123456789",
>     "tenant_id": "123456789",
>     "name": "YM_sample_sg",
>     "description": "sg Description",
>     "security_group_rules": [
>         {
>             "id": "8f01055a-eea5-407f-85a1-e1fcee5680e4",
>             "project_id": "",
>             "tenant_id": "123456789",
>             "name": "",
>             "description": "",
>             "security_group_id": "",
>             "remote_group_id": null,
>             "direction": "egress",
>             "remote_ip_prefix": null,
>             "protocol": null,
>             "port_range_max": null,
>             "port_range_min": null,
>             "ethertype": "IPv4"
>         },
>         {
>             "id": "cfbb758a-224d-42fc-8099-fa0e9df30969",
>             "project_id": "",
>             "tenant_id": "123456789",
>             "name": "",
>             "description": "",
>             "security_group_id": "",
>             "remote_group_id": null,
>             "direction": "egress",
>             "remote_ip_prefix": null,
>             "protocol": null,
>             "port_range_max": null,
>             "port_range_min": null,
>             "ethertype": "IPv6"
>         }
>     ],
>     "create_at": "2022-09-19 23:13:58",
>     "update_at": "2022-09-19 23:13:58"
> }
> ```

I can create sg without assigned ID on .77 and it will return a new ID to me:

```
curl -X POST --header "Content-Type: application/json" --header "Accept: */*" -d "{
  \"security_group\": {
    \"description\": \"string\",
    \"name\": \"sg1\",
    \"project_id\": \"3dda2801-d675-4688-a63f-dcda8d327f50\",
    \"tenant_id\": \"3dda2801-d675-4688-a63f-dcda8d327f50\"
  }
}" "http://10.213.43.77:30008/project/3dda2801-d675-4688-a63f-dcda8d327f50/security-groups"
```
Response Body:

```
{
  "security_group": {
    "id": "66345a01-98fb-4ced-9d66-ae849d55c912",
    "project_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
    "tenant_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
    "name": "sg1",
    "description": "string",
    "security_group_rules": [
      {
        "id": "98334775-648d-4cda-b8d7-43aefb855b1a",
        "project_id": null,
        "tenant_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
        "name": null,
        "description": null,
        "security_group_id": "66345a01-98fb-4ced-9d66-ae849d55c912",
        "remote_group_id": null,
        "direction": "egress",
        "remote_ip_prefix": null,
        "protocol": null,
        "port_range_max": null,
        "port_range_min": null,
        "ethertype": "IPv4"
      },
      {
        "id": "5b274be4-b6fa-489a-b668-abde3f75730e",
        "project_id": null,
        "tenant_id": "3dda2801-d675-4688-a63f-dcda8d327f50",
        "name": null,
        "description": null,
        "security_group_id": "66345a01-98fb-4ced-9d66-ae849d55c912",
        "remote_group_id": null,
        "direction": "egress",
        "remote_ip_prefix": null,
        "protocol": null,
        "port_range_max": null,
        "port_range_min": null,
        "ethertype": "IPv6"
      }
    ],
    "create_at": "2022-09-20 00:47:02",
    "update_at": "2022-09-20 00:47:02"
  }
}
```

**Comment by yanmo96** [comment]
> 

Yes, Alcor will give a sg ID. The issue earlier was, I didn't specify an ID for it, but ID is a field in the struct, so it put "" as the ID. Will put a fix to it. Thanks

File: services/merak-network/activities/vnet_create_activity.go (hunk 2 above)
Comment (yanmo96) [reply services/merak-network/activities/vnet_create_activity.go]: removed now

**Comment by yanmo96** [comment]
@cj-chung SG ID is not assigned anymore, and the above mention of cleaning the code has been done, with similar cleaning in other places too. Please take a look. Thanks

**Review by cj-chung — APPROVED** [review]
LGTM

**Pull request merged**